3p/ampcontext.js

```diff
@@ -138,4 +148,21 @@ export class AmpContext {
     }
     return ancestors[(ancestors.length - 1) - depth];
   }
+
+  /**
+   *  Checks to see if there is a window variable assigned with the
+   *  sentinel value, sets it, and returns true if so.
+   *  @private
+   */
+  hasContextDataOnWindow_() {
+    if (!this.win_.AMP_CONTEXT_DATA) {
+      return false;
+    }
+    if (typeof this.win_.AMP_CONTEXT_DATA == 'string') {
+      this.sentinel = this.win_.AMP_CONTEXT_DATA;
+    } else {
+      this.setupMetadata_(this.win_.AMP_CONTEXT_DATA);
```

Comment:
now this method mutates states, so we should rename it.
And probably you will just want to move line 33 into this method as well.